metrics-core/src/main/java/com/codahale/metrics/MetricRegistry.java

```diff
@@ -301,6 +301,19 @@ public boolean isInstance(Metric metric) {
         });
     }
 
+    /**
+     * Return the {@link Gauge} registered under this name; or create and register
+     * a new {@link SettableGauge} if none is registered.
+     *
+     * @param name the name of the metric
+     * @return a new or pre-existing {@link SettableGauge}
+     * @since 4.2
+     */
+    @SuppressWarnings("unchecked")
+    public <T> SettableGauge<T> gauge(String name) {
```

Comment:
Shouldn't line 313 be
public <T> SettableGauge<T> settableGauge(String name) { ?

good to call it a few times in the test as well.

Reply:
I'd like to keep symmetry with `timer(String)` and `timer(String, Supplier)` in this case.

@dropwizard/metrics @dropwizard/committers Any opinions on the name?

Reply:
Myabe I'm missing something but comment is on the name of the created method.   Shouldn't it be
settableGauge( String name ) to distinguish it from the already existing gauge( String name ) ?   If java
will automatically call the proper one depending on the lefthand side of user's assignment, then my
comment is moot (excpet for the need for a few test cases calls).

Reply:
> distinguish it from the already existing gauge( String name )

@jg167 There is no such method. 😉

Reply:
ah, my bad.  I was thinking that there was one (i.e. just like timer(), coutner()...)
still maybe gauge( Name ) creating a SettableGauge instead of a regular Gauge is a bit confusing? 

plus a few calls of whatever is used in the test is needed (I see a coverage checkk did not pass, I presume due to that).

Reply:
>  still maybe gauge( Name ) creating a SettableGauge instead of a regular Gauge is a bit confusing?

A `SettableGauge` instance can be used just as a normal `Gauge` if you're only reading its value.

Otherwise the `MetricRegistry#gauge(String)` method doesn't make sense in the first place.

Reply:
Perhaps I'm missing something here but I thought that a normal Gauge has a supplier and every time you call getValue() that supplier is called to actually get the value.  For a SettableGauge there is no supplier and the value stored in the SettableGauge by the last setValue() is returned.

Reply:
@jg167 Let's say there was a method `MetricRegistry#settableGauge(String)`, what would it return if the name of a normal `Gauge<T>` was provided?

It would either have to throw an exception (booo!) or return `null` which makes it fundamentally different from the behavior of other factory methods such as `MetricRegistry#timer(String)` or `MetricRegistry#gauge(String, Supplier)` which will _never_ return `null`.

Reply:
I woudl sayt it should create a SettableGauge with the same name as the Gauge.  perhpas that is a problem for the implementation but they are two different things.  A gauge has a supplier function and no storage while a settableGauge has storage and no suppier function.     also there is no convience function gague( <T>, supplier), are you proposging to add that as well?   what would such a fucntion return if the name matched but the supllier function did not?

Reply:
@joschi , since you have merged this into release/4.2.x any idea when a new beta will be generaetd?  -thanks.

Reply:
> also there is no convience function gague( , supplier), are you proposging to add that as well?

[`MetricRegistry#gauge(String name, MetricSupplier<Gauge> supplier)`](https://javadoc.io/static/io.dropwizard.metrics/metrics-core/4.1.18/com/codahale/metrics/MetricRegistry.html#gauge-java.lang.String-com.codahale.metrics.MetricRegistry.MetricSupplier-)

Reply:
> since you have merged this into release/4.2.x any idea when a new beta will be generaetd?

We'll release another beta probably after #1806 has been reviewed and merged.